plugins/source/metrics.go

```diff
@@ -123,3 +130,51 @@ func (s *Metrics) TotalResourcesAtomic() uint64 {
 	}
 	return total
 }
+
+func (s *Metrics) MarkStart(table *schema.Table, clientID string) {
+	s.TableClient[table.Name][clientID].Mutex.Lock()
```

Comment:
We might want to do the start time calculation before the lock, WDYT? That would be more accurate to when the method was called.

Same for end time calculation (and maybe even pass that value to relations)